File: src/main/java/pl/pkowalczyk/carparts/demo/model/OrderDetail.java
package pl.pkowalczyk.carparts.demo.model;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Setter
@Getter
@NoArgsConstructor
@Entity
@Table(name = "order_detail")
public class OrderDetail {
    @Id
    @Column(name = "order_detail_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "part_id")
    private Part part;
    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;
}

File: user-app/src/app/carparts/OrderDetail.ts
import {Part} from './Part';
import {Order} from './Order'

export class OrderDetail  {
	id: number;
	part: Part;
	order: Order;


	constructor(id: number, part: Part, order: Order) {
		this.id = id;
		this.part = part;
		this.order = order;
	}
}